contrib/jackc/pgx/trace.go

```diff
@@ -0,0 +1,63 @@
+package pgx
+
+import (
+	"context"
+	"math"
+	"time"
+
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace"
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace/ext"
+	ddtracer "gopkg.in/DataDog/dd-trace-go.v1/ddtrace/tracer"
+
+	"github.com/jackc/pgx/v5"
+)
+
+// tracer contains configs to tracing and would implement QueryTracer, BatchTracer,
+// CopyFromTracer, PrepareTracer and ConnectTracer from pgx.
+type tracer struct {
+	serviceName   string
+	tags          map[string]interface{}
+	analyticsRate float64
+	traceArgs     bool
+	traceStatus   bool
+}
+
+// TraceQueryStart marks the start of a query, implementing pgx.QueryTracer
+func (t *tracer) TraceQueryStart(ctx context.Context, _ *pgx.Conn, data pgx.TraceQueryStartData) context.Context {
+	opts := []ddtrace.StartSpanOption{
+		ddtracer.ServiceName(t.serviceName),
+		ddtracer.SpanType(ext.SpanTypeSQL),
+		ddtracer.StartTime(time.Now()),
+		ddtracer.Tag("sql.query_type", "Query"),
+		ddtracer.Tag(ext.ResourceName, data.SQL),
```

Comment:
```suggestion
		ddtracer.Tag(ext.ResourceName, data.SQL),
		ddtracer.Tag(ext.Component, "jackc/pgx"),
		ddtracer.Tag(ext.SpanKind, ext.SpanKindClient),
```

Component and SpanKind tags are necessary for all components